federation/pkg/federation-controller/sync/controller.go

```diff
@@ -384,6 +391,50 @@ func (s *FederationSyncController) delete(obj pkgruntime.Object, kind string, na
 	return nil
 }
 
+// updateFedSpec updates the spec against the federated object using the adapters UpdateFedSpecHook of need be.
+func (s *FederationSyncController) updateFedSpec(fedObj pkgruntime.Object, hook federatedtypes.UpdateFedSpecHook) reconcileStatus {
```

Comment:
It's not clear to me why this method should want to iterate over every cluster object for every federation object.  Please provide an explanation of why this approach is necessary, with pointers to docs/code as appropriate.